pkg/kubelet/kubelet_pods.go

```diff
@@ -1458,6 +1458,11 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		switch cs.State {
 		case kubecontainer.ContainerStateRunning:
 			status.State.Running = &v1.ContainerStateRunning{StartedAt: metav1.NewTime(cs.StartedAt)}
+			// Use the ready value from previousStatus so that we don't initialize it to false
```

Comment:
I'm confused by this. Isn't this change just delaying setting the status for 1 cycle in all cases? I don't think we want that.

Reply:
Since kubelet always initialize status.Ready to false, all `ready` pods are marked `not ready` after kubelet restart, those pods are then kicked from Endpoints. This code keep the value of status.Ready received from api-server so that it will not be set to `false` until probe worker finishes the test.